iceoryx_hoofs/test/moduletests/test_cxx_string.cpp

```diff
@@ -2059,12 +2059,65 @@ TYPED_TEST(stringTyped_test, ConcatenateStringLiteralAndStringWorks)
 TEST(StringLiteralConcatenation, ConcatenateOnlyStringLiteralsWorks)
 {
     ::testing::Test::RecordProperty("TEST_ID", "f591fb69-a8fa-4d8e-b42d-29573b4a20f7");
-    auto testString = iox::cxx::concatenate("Ferdi", "nandSpitzschnu", "ef", "fler");
+    auto testString = concatenate("Ferdi", "nandSpitzschnu", "ef", "fler");
     EXPECT_THAT(testString.capacity(), Eq(25U));
     EXPECT_THAT(testString.size(), Eq(25U));
     EXPECT_THAT(testString.c_str(), StrEq("FerdinandSpitzschnueffler"));
 }
 
+TYPED_TEST(stringTyped_test, ConcatenateEmptyStringAndNullCharReturnsStringWithSizeOne)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "6a4374fb-b21f-4465-95dc-22b898079c81");
+    using MyString = typename TestFixture::stringType;
+    constexpr auto STRINGCAP = MyString::capacity();
+    auto result = concatenate(this->testSubject, '\0');
```

Comment:
Since the types of the operands are not equal, consider adding a test where the operands are swapped. Could be done in the same test if we have faith (or check) that `this->testSubject` was not changed.

I know that the combinatorics get out of hand fast, but here it might be reasonable and feasible.